### ISSUE TITLE
Loss function of poisoning data not found

### ISSUE THREAD

**Issue by saulchiu** [IBM/BadDiffusion#13]
if loss_type == 'l1':
        loss = F.l1_loss(target, predicted_noise)
    elif loss_type == 'l2':
        loss = F.mse_loss(target, predicted_noise)
    elif loss_type == "huber":
        loss = F.smooth_l1_loss(target, predicted_noise)

I see you user L-2 to optimize the noice predictor, which is as same as paper.
But i could not find the loss function of poisoning data.
<img width="455" alt="截屏2024-04-21 18 54 50" src="https://github.com/IBM/BadDiffusion/assets/147218976/02ccd7b7-0137-4e60-bd5f-8c72d562589b">

**Comment by FrankCCCCC** [comment]
Hi, thank you for your interest. The loss function for the poisoning data is at Line 257, the function ``q_sample_diffuser``. The ``target`` in Line 293 is $\frac{\rho_t \sigma_t}{1-\alpha_t} \mathbf{r} + \epsilon$ and the ``x_noisy`` is $\mathbf{x}_{t}'(\mathbf{y}, \mathbf{y}, \epsilon)$. Let me know if you have further questions.

**Comment by saulchiu** [comment]
Oh I see, it is kind of you, thank you very much. This is really helpful to me :)
